plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -193,6 +200,64 @@ def _state_deleted(self, want, have):
                     commands = update_states(diff_want, "deleted")
         return commands, requests
 
+    def _state_replaced(self, want, have, diff):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        requests = []
+        commands = []
+
+        diff1 = get_diff(have, want, TEST_KEYS, is_skeleton=True)
+        if diff1:
+            requests = self.get_delete_all_mclag_domain_request(have)
+            if len(requests) > 0:
+                send_requests(self._module, requests)
+            else:
+                commands = []
+
+        exist_mclag_facts = self.get_mclag_facts()
```

Comment:
Don't get the "facts" again or re-calculate the diffs. It is a costly operation, and the set of commands to apply is already known from the previous "diff" result.

Reply:
Refer previous comment.